src/lib.rs

```diff
@@ -12,6 +12,14 @@
 //! This is very similar to how `futures::stream::unfold` creates a `Stream`
 //! implementation from a seed value and a future-returning closure.
 //!
+//! # Error behaviour
+//!
+//! If any of the [`Sink`] methods produces an error, the sink transitions to
+//! a closed state. Invoking [`Sink::poll_ready`] or [`Sink::start_send`]
```

Comment:
```suggestion
//! a closed state. Invoking [`Sink::poll_ready`]
```

Small nit: I would not mention `start_send` here, as `start_send` doesn't panic due to a previous error, but due to the sink being in the wrong state.

Reply:
The sink is in a wrong state as a result of an error. `poll_ready` also panics because the sink is in a wrong state. The state transition is an implementation detail to handle failures. Of course clients are supposed to call `poll_ready` before `start_send` but I like to mention all API methods and clearly state that `poll_flush` and `poll_close` will not panic, whereas the other two will if an error happened.

Reply:
Sounds good. Mentioning `start_send` again doesn't hurt.